docs/swarm_at_scale/deploy-app.md

```diff
@@ -0,0 +1,425 @@
+    -e constraint:com.function==worker01 \
+    --net="voteapp" \
+    --net-alias=workers \
+    --name worker01 docker/example-voting-app-worker
+    ```
+
+6. Start the results application.
+
+    ```bash
+    $ docker -H $(docker-machine ip manager):3376 run -t -d \
+    -p 80:80 \
+    --label=interlock.hostname=results \
+    --label=interlock.domain=myenterprise.com \
+    -e constraint:com.function==dbstore \
+    --net="voteapp" \
+    --name results-app docker/example-voting-app-result-app
+    ```
+
+7. Start voting application twice, on each frontend node.
+
+    ```bash
+    $ docker -H $(docker-machine ip manager):3376 run -t -d \
+    -p 80:80 \
+    --label=interlock.hostname=vote \
+    --label=interlock.domain=myenterprise.com \
+    -e constraint:com.function==frontend01 \
+    --net="voteapp" \
+    --name voting-app01 docker/example-voting-app-voting-app
+    ```
+
+    And again on the other frontend node.
+
+    ```bash
+    $ docker -H $(docker-machine ip manager):3376 run -t -d \
+    -p 80:80 \
+    --label=interlock.hostname=vote \
+    --label=interlock.domain=myenterprise.com \
+    -e constraint:com.function==frontend02 \
+    --net="voteapp" \
+    --name voting-app02 docker/example-voting-app-voting-app
+    ```
+
+
+## Task 3. Check your work and update /etc/hosts
+
+In this step, you check your work to make sure the Nginx configuration recorded
+the containers correctly. You'll update your local systems `/etc/hosts` file to
+allow you to take advantage of the loadbalancer.
+
+1. Change to the `loadbalancer` node.
+
+    ```bash
+    $ eval $(docker-machine env loadbalancer)
+    ```
+2. Check your work by reviewing the configuration of nginx.
+
+    ```html
+    $ docker exec interlock cat /etc/conf/nginx.conf
+    ... output snipped ...
+
+    upstream results.myenterprise.com {
+        zone results.myenterprise.com_backend 64k;
+
+        server 192.168.99.111:80;
+
+    }
+    server {
+        listen 80;
+
+        server_name results.myenterprise.com;
+
+        location / {
+            proxy_pass http://results.myenterprise.com;
+        }
+    }
+    upstream vote.myenterprise.com {
+        zone vote.myenterprise.com_backend 64k;
+
+        server 192.168.99.109:80;
+        server 192.168.99.108:80;
+
+    }
+    server {
+        listen 80;
+
+        server_name vote.myenterprise.com;
+
+        location / {
+            proxy_pass http://vote.myenterprise.com;
+        }
+    }
+
+    include /etc/conf/conf.d/*.conf;
+    }
+    ```
+
+  The `http://vote.myenterprise.com` site configuration should point to either
+  frontend node. Requests to `http://results.myenterprise.com` go just to the
+  single `dbstore` node where the `example-voting-app-result-app` is running.
+
+8. On your local host, edit `/etc/hosts` file add the resolution for both these
+sites.
+
+9. Save and close the `/etc/hosts` file.
+
+10. Stop and restart the `nginx` container so it picks up the new `/etc/hosts`
+configuration.
```

Comment:
nginx doesn't need `/etc/host` file. We restart it to reload its configuration because interlock doesn't force configuration reload, which is a bug.